File: frontend/src/api/costs.ts
import { api } from "./client";

export interface CostRow {
  key: string;
  calls: number;
  tokens: number;
  cost_usd: number;
}

export interface CostDay {
  day: string;
  cost_usd: number;
  tokens: number;
}

export interface CostSummary {
  total_cost_usd: number;
  total_tokens: number;
  total_calls: number;
  month_cost_usd: number;
  month_tokens: number;
  month_calls: number;
  input_rate_per_1m: number;
  output_rate_per_1m: number;
  by_day: CostDay[];
  by_user: CostRow[];
  by_model: CostRow[];
}

export const getCostSummary = () => api.get<CostSummary>("/api/costs/summary");

File: backend/app/schemas/costs.py
from pydantic import BaseModel


class CostRow(BaseModel):
    key: str
    calls: int
    tokens: int
    cost_usd: float


class CostDay(BaseModel):
    day: str
    cost_usd: float
    tokens: int


class CostSummary(BaseModel):
    total_cost_usd: float
    total_tokens: int
    total_calls: int
    month_cost_usd: float
    month_tokens: int
    month_calls: int
    input_rate_per_1m: float
    output_rate_per_1m: float
    by_day: list[CostDay]
    by_user: list[CostRow]
    by_model: list[CostRow]
